src/en/clouds.md

```diff
@@ -311,81 +320,136 @@ expand the relevant section). You can also generate a YAML file.
+### Adding clouds manually
+
+As an alternative to the interactive method, clouds can be added manually. This
+can assist with automation.
+
+The manual method necessitates the use of a [YAML-formatted][yaml]
+configuration file. It has the following format:
 
-In this case it is possible to create
-a [YAML][yaml] formatted file with the information Juju requires and import this
-new definition. The file should follow this general format:
-  
 ```yaml
 clouds:
   <cloud_name>:
     type: <type_of_cloud>
-    auth-types: <[access-key, oauth, userpass]>
+    auth-types: [<authenticaton_types>]
     regions:
       <region-name>:
         endpoint: <https://xxx.yyy.zzz:35574/v3.0/>
 ```
-with the releavant values substituted in for the parts indicated
-(within '<' '>').
 
-For example, a typical OpenStack cloud on the local network you want to call 
-'mystack' would appear something like this:
-  
+The table below shows the authentication types available for each cloud type.
+It does not include the `interactive` type as it does not apply in the context
+of adding a cloud manually.
+
+| cloud type      | authentication types                      |
+|----------------|------------------------------------------|
+`azure`		  | `service-principal-secret`
+`cloudsigma`	  | `userpass`
+`ec2`		  | `access-key`
+`gce`		  | `jsonfile,oauth2`
+`joyent`	  | `userpass`
+`lxd`		  | n/a
+`maas`		  | `oauth1`
+`manual`	  | n/a
+`openstack`	  | `access-key,userpass`
+`oracle`	  | `userpass`
+`rackspace`	  | `userpass`
+`vsphere`	  | `userpass`
+
+To add a cloud in this way we simply supply an extra argument to specify the
+relative path to the file:
+ 
+`juju add-cloud <cloud-name> <cloud-file>`
+
+To confirm that the cloud has been successfully added you can re-run the
+`clouds` command.
+
+Below we provide two examples.
+
+#### Manually adding MAAS clouds
+
+This example covers manually adding a MAAS cloud to Juju. It also demonstrates
+how multiple clouds of the same type can be defined and added.
+
+Here is the YAML file:
+
+```yaml
+clouds:
+   devmaas:
+      type: maas
+      auth-types: [oauth1]
+      endpoint: http://devmaas/MAAS
+   testmaas:
+      type: maas
+      auth-types: [oauth1]
+      endpoint: http://172.18.42.10/MAAS
+   prodmaas:
+      type: maas
+      auth-types: [oauth1]
+      endpoint: http://prodmaas/MAAS
+```
+
+This defines three MAAS clouds and refers to them by their respective
+region controllers.
+
+To add clouds 'devmaas' and 'prodmaas', assuming the configuration file is
+`maas-clouds.yaml` in the current directory, we would run:
+
+```bash
+juju add-cloud devmaas maas-clouds.yaml
+juju add-cloud prodmaas maas-clouds.yaml
+```
+
+You must now add a credential for this cloud prior to creating a controller
+(`juju bootstrap`). See the [Credentials][credentials] page for details. The
+[Using MAAS with Juju][clouds-maas-add-credentials] page also includes
+MAAS-specific guidance on this.
+
+#### Manually adding an OpenStack cloud
```

Comment:
This section should very definitely include a link to the how to for setting up YOUR OWN openstack cloud. Perhaps in the form of a note at the beginning (These instructions assume an Openstack cloud which already has relevant Ubuntu images available. For help in configuring your own Openstack cloud to work with Juju, see...) or something

Reply:
Agreed.

Reply:
did you do this?

Reply:
Not yet.